### PR TITLE
Container-port flag is deprecated

### PR DESCRIPTION
**Pull request by adohe-zz**
Refer #25039 for more detail. @kargakis ptal. Also in this PR just mark this flag as deprecated, we should also remove the code in the future.

### CODE REVIEW

**Comment by kargakis** [comment]
LGTM

@smarterclayton @bgrant0607 WDYT?

**Comment by AdoHe** [comment]
@k8s-bot test this: #23974

**Comment by AdoHe** [comment]
@k8s-bot test this issue: #23974

**Comment by smarterclayton** [comment]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 - [v1.3.0-alpha.3](#v130-alpha3)
   - [Downloads](#downloads)
+- [v1.3.0-alpha.3](#v130-alpha3)
```
Comment (luxas) [inline CHANGELOG.md]: Why is this added?

**Comment by luxas** [comment]
Is the policy to add `release-note` to deprecations? I think we should do that.

**Comment by luxas** [comment]
Okay, it's worthy a release note as it changes the ux. cc/ @david-mcmahon 
Is it okay if I'm adding back `lgtm`, given the multiple ones already?

**Comment by bgrant0607** [comment]
Thanks, LGTM.

Deprecations belong in the "action required" section of the release notes. I updated that label.
